workflows/cloudify_system_workflows/deployment_update/workflow.py

```diff
@@ -68,6 +70,41 @@ def create_steps(*, update_id):
     )
 
 
+def _modified_attr_nodes(steps):
+    """Based on the steps, find what attributes changed for every node.
+
+    Returns a dict of {node-id: [attributes that changed]}.
+    """
+    modified = defaultdict(list)
+    modified_entity_type = {
+        'plugin': 'plugins',
+        'relationship': 'relationships',
+        'property': 'properties'
+    }
+    for step in steps:
+        parts = step['entity_id'].split(':')
+        node_id = parts[1]
+        entity_type = step['entity_type']
+        if entity_type in modified_entity_type:
+            modified[node_id].append(modified_entity_type[entity_type])
+        if entity_type == 'operation':
+            if 'relationships' in parts:
+                modified[node_id].append('relationships')
+            else:
+                modified[node_id].append('operations')
```

Comment:
So if entity_type is not in [plugin, relationship, property, operation] then we silently fall through? Should we perhaps scream instead?

Reply:
no, this implementation is not complete yet